pkg/actors/actors.go

```diff
@@ -158,20 +162,20 @@ func NewActors(opts ActorsOpts) ActorRuntime {
 
 func newActorsWithClock(opts ActorsOpts, clock clock.WithTicker) ActorRuntime {
 	a := &actorsRuntime{
-		appChannel:           opts.AppChannel,
-		grpcConnectionFn:     opts.GRPCConnectionFn,
-		actorsConfig:         opts.Config,
-		timers:               timers.NewTimersProvider(clock),
-		tracingSpec:          opts.TracingSpec,
-		resiliency:           opts.Resiliency,
-		storeName:            opts.StateStoreName,
-		placement:            opts.MockPlacement,
-		actorsTable:          &sync.Map{},
-		clock:                clock,
-		internalActors:       map[string]InternalActor{},
-		internalActorChannel: newInternalActorChannel(),
-		compStore:            opts.CompStore,
-		sec:                  opts.Security,
+		appChannel:         opts.AppChannel,
+		grpcConnectionFn:   opts.GRPCConnectionFn,
+		actorsConfig:       opts.Config,
+		timers:             timers.NewTimersProvider(clock),
+		tracingSpec:        opts.TracingSpec,
+		resiliency:         opts.Resiliency,
+		storeName:          opts.StateStoreName,
+		placement:          opts.MockPlacement,
+		actorsTable:        &sync.Map{},
+		clock:              clock,
+		internalActorTypes: haxmap.New[string, InternalActorFactory](4), // Initial capacity should be enough for the current built-in actors
+		internalActors:     haxmap.New[string, InternalActor](32),
```

Comment:
Magic number?

Reply:
It's just an initial capacity that makes sense. If more capacity is needed, it'll be allocated